### PR TITLE
Fix lighthouse navigate to a wrong URL

### PR DESCRIPTION
**Pull request by CSalih**
## Hey, I just made a Pull Request!

This PR will fix the issue #15231 where audit list item and create audit button navigated to a wrong URL.

With `resolvePath(generatePath('audit/:id', { id: audit.id }), '../../')` we navigate from `/<base-url>/audit/<id>` to `/audit/<id>` but this is wrong because the base path will be removed. Therefore we use `useRouteRef`, this will navigate to the right url with respect to the base path.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/lighthouse/src/components/AuditView/index.test.tsx [hunk 1]
```diff
@@ -149,6 +149,28 @@ describe('AuditView', () => {
...
+        ),
+      );
+
+      await rendered.findByTestId('audit-sidebar');
+
+      websiteResponse.audits.forEach(a => {
+        expect(
+          rendered.getByText(formatTime(a.timeCreated)).parentElement
+            ?.parentElement,
+        ).toHaveAttribute('href', `${basePath}/audits/${a.id}`);
```
Comment (CSalih) [inline plugins/lighthouse/src/components/AuditView/index.test.tsx:171]: For me: this should be `${basePath}/audit/${a.id}`

**Comment by freben** [comment]
Just a gentle nudge here. Note the test errors!

**Comment by CSalih** [comment]
Hi @freben , 
thank you for you Feedback! 
I was not able to set the the applications base path to e.g. `/lighthouse` for this one test.  I will search tomorrow a solution or do you maybe know a way how to do that?

**Comment by freben** [comment]
You should be able to supply a config that has that setting. Your `ApiProvider` could contain a `configApiRef` that's like `new ConfigReader({ app: { basePath: '/something' } })`

**Comment by CSalih** [comment]
@freben Sorry for the delay! It took me a while to fix this test case, thank you so much for your great help. 
This is my first contribution to backstage :rocket: , be as strict with the review as necessary

**Comment by freben** [comment]
(just popping in to mention I'm off for the holidays and reviews might take a while longer)

**Comment by CSalih** [comment]
just resetting the stale status

**Review by freben — APPROVED** [review]
Thanks!

**Pull request merged**